### PR TITLE
Add total times to story and display them

### PR DESCRIPTION
**Pull request by sarah256**
Looks like:
screencapture-localhost-4200-advisory-39082-2019-07-08-11_47_56
screencapture-localhost-4200-advisory-39082-2019-07-08-11_48_12

### CODE REVIEW

**Comment by gnaponie** [comment]
How does it look like when it's seen from mobile or with smaller screen/window?

File: src/app/story/story.component.ts [hunk 1]
```diff
@@ -20,6 +21,9 @@ export class StoryComponent implements OnInit, AfterViewInit, OnDestroy {
   loading: boolean;
   story: StoryAPI;
   selectedNode: any;
+  totalLeadTime: string;
```
Comment (gnaponie) [inline src/app/story/story.component.ts]: That's my ignorance... but what does "lead" mean in this case?

File: src/app/pipes/timedisplay.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
+import { PipeTransform, Pipe } from '@angular/core';
+
+
+@Pipe({name: 'timeDisplay'})
+export class TimeDisplayPipe implements PipeTransform {
+    transform(timeInSeconds: number): string {
```
Comment (gnaponie) [inline src/app/pipes/timedisplay.ts]: There's no available library function to calculate days/hours/minutes?

**Comment by gnaponie** [comment]
> How does it look like when it's seen from mobile or with smaller screen/window?

Oh ok. Ignore this comment. I saw in the code it just gets hidden. It looks good!

File: src/app/story/story.component.ts (hunk 1 above)
Comment (sarah256) [reply src/app/story/story.component.ts]: It means total time it takes to get through the pipeline. This was the name Ralph used in the MVP, so just wanted to keep it consistent :smile:

File: src/app/pipes/timedisplay.ts (hunk 2 above)
Comment (sarah256) [reply src/app/pipes/timedisplay.ts]: Not that I could find sadly. We also wanted to have it show hours/minutes/seconds if it was less than a day, minutes and seconds if less than an hour, and then only seconds if less than a minute, so this seemed easier to just customize.

File: src/app/story/story.component.ts (hunk 1 above)
Comment (sarah256) [reply src/app/story/story.component.ts]: Do you think it'd be helpful to leave a comment here explaining that?

**Comment by sarah256** [comment]
This is blocked until [this PR](https://github.com/release-engineering/estuary-api/pull/254) in the API is merged.

**Comment by sarah256** [comment]
Now that [the API PR](https://github.com/release-engineering/estuary-api/pull/254) is merged, we can open this PR to review again.

**Comment by mprahl** [comment]
@sarah256 could you please resolve the conflicts. I'll review this next week.

**Comment by sarah256** [comment]
This was working, but since rebasing it has stopped working. :disappointed: Will try to fix today

**Comment by sarah256** [comment]
@mprahl conflicts resolved, should be ready to review again! :smile:

File: src/app/story/storysidebar/storysidebar.component.html [hunk 3]
```diff
@@ -1,45 +1,65 @@
...
-      <!-- *ngIf is used because it should only tooltip on long UIDs -->
-      <h2 *ngIf="(node.display_name | nodeDisplayName).length > 30; else shortUid" tooltip="{{ node.display_name | nodeDisplayName }}">
-        {{ node.display_name | nodeDisplayName | truncate: 30 }}
-      </h2>
-      <ng-template #shortUid>
-        <h2>{{ node.display_name | nodeDisplayName }}</h2>
-      </ng-template>
-    </a>
+  <div class="timeline-properties mobile-hide" (click)="sidebarOpen = !sidebarOpen; $event.preventDefault()">
     <table class="sidebar-properties">
```
Comment (mprahl) [inline src/app/story/storysidebar/storysidebar.component.html]: Clicking on this table causes the sidebar to collapse and expand. That shouldn't be happening.

File: src/app/story/story.component.css [hunk 4]
```diff
@@ -2,10 +2,15 @@
...
 }
 
 @media (min-width: 992px) {
     .diagram {
         margin: 2rem 0 0 3rem;
     }
 }
+
+.timeline-properties {
+    float: right;
```
Comment (mprahl) [inline src/app/story/story.component.css]: This doesn't do anything since it's in the wrong CSS file.

File: src/app/story/storysidebar/storysidebar.component.html [hunk 5]
```diff
@@ -1,45 +1,65 @@
...
-    <a class="sidebar__external-system-link" [href]="node | nodeExternalUrl" target="_blank">
-      <!-- *ngIf is used because it should only tooltip on long UIDs -->
-      <h2 *ngIf="(node.display_name | nodeDisplayName).length > 30; else shortUid" tooltip="{{ node.display_name | nodeDisplayName }}">
-        {{ node.display_name | nodeDisplayName | truncate: 30 }}
-      </h2>
-      <ng-template #shortUid>
-        <h2>{{ node.display_name | nodeDisplayName }}</h2>
-      </ng-template>
-    </a>
+  <div class="timeline-properties mobile-hide" (click)="sidebarOpen = !sidebarOpen; $event.preventDefault()">
```
Comment (mprahl) [inline src/app/story/storysidebar/storysidebar.component.html]: When the width is small enough, this causes the text to float over the story:
image

Comment (mprahl) [inline src/app/story/storysidebar/storysidebar.component.html]: When this was added, the animation of the sidebar opening and closing looks off.

[the remark this answers is not in the document]
Comment (sarah256) [reply src/app/story/storysidebar/storysidebar.component.html]: What would you suggest as a fix? If it is not in the sidebar that makes it hard for it to respond to the sidebar changing size

[the remark this answers is not in the document]
Comment (mprahl) [reply src/app/story/storysidebar/storysidebar.component.html]: @sarah256 I'm personally fine if the sidebar covers up the time statistics if the screen is too small. So you could just make a separate component for the time statistics and place it next to the story.

[the remark this answers is not in the document]
Comment (sarah256) [reply src/app/story/storysidebar/storysidebar.component.html]: @mprahl When the screen is too small, the properties are hidden. I think maybe we can discuss in person?

Comment (sarah256) [reply src/app/story/storysidebar/storysidebar.component.html]: Is this what you were referring to in your previous comment?

**Comment by sarah256** [comment]
@mprahl This should be ready for review again

File: src/app/story/story.component.html [hunk 6]
```diff
@@ -5,12 +5,13 @@
   <div class="diagram" id="js-diagram">
     <ng-container *ngIf="story">
       <app-storyrow [node]="node" [backwardSiblings]="story.meta.story_related_nodes_backward[i]"
-          [forwardSiblings]="story.meta.story_related_nodes_forward[i]" [last]="last"
-          [active]="node === selectedNode"
-          *ngFor="let node of story.data; let i = index; let last = last">
+          [forwardSiblings]="story.meta.story_related_nodes_forward[i]" [first]="first" [last]="last"
+          [active]="node === selectedNode" [totalLeadTime]="story.meta.total_lead_time"
```
Comment (mprahl) [inline src/app/story/story.component.html]: Optional: Rather than 3 properties, do you think it makes sense to just pass in `story.meta` as a single property into `StoryRowComponent`?

File: src/app/story/story.component.ts [hunk 7]
```diff
@@ -13,16 +14,17 @@ declare var jsPlumb: jsPlumbInstance;
...
 export class StoryComponent implements OnInit, AfterViewInit, OnDestroy {
 
   loading: boolean;
   story: StoryAPI;
   selectedNode: any;
+  sidebarOpen: boolean;
 
   constructor(private storyService: StoryService, private route: ActivatedRoute,
-              private elRef: ElementRef) {
+              private elRef: ElementRef, private notification: NotificationService) {
```
Comment (mprahl) [inline src/app/story/story.component.ts]: Could you please remove this? It's not used as far as I can tell.

File: src/app/story/storyrow/storyrow.component.html [hunk 8]
```diff
@@ -65,3 +65,10 @@
     </div>
   </div>
 </div>
+<div class="node-siblings-column story-row-column total-time-column" *ngIf="first">
+  <app-totaltimes [ngClass]="'total-times'"
```
Comment (mprahl) [inline src/app/story/storyrow/storyrow.component.html]: For long HTML elements like this, the formatting would normally be:
```html
<app-totaltimes
  [ngClass]="'total-times'"
  [totalLeadTime]="totalLeadTime"
  [totalProcessingTime]="totalProcessingTime"
  [totalWaitTime]="totalWaitTime"
>
```

With that being said, I don't think we follow that rule in this code base.

Comment (mprahl) [inline src/app/story/storyrow/storyrow.component.html]: Could you use `class="total-times"`? I don't think `ngClass` is necessary since it's a static value.

File: src/app/story/storyrow/totaltimes/totaltimes.component.html [hunk 9]
```diff
@@ -0,0 +1,18 @@
+<div class="timeline-properties mobile-hide" (click)="sidebarOpen = !sidebarOpen; $event.preventDefault()">
```
Comment (mprahl) [inline src/app/story/storyrow/totaltimes/totaltimes.component.html]: `timeline-properties` is not used as far as I can tell. Could you please remove it?

File: src/app/story/storyrow/totaltimes/totaltimes.component.ts [hunk 10]
```diff
@@ -0,0 +1,19 @@
+import { Component, Input } from '@angular/core';
+
+
+@Component({
+  selector: 'app-totaltimes',
+  templateUrl: './totaltimes.component.html',
+  styleUrls: ['./totaltimes.component.css', '../../storysidebar/storysidebar.component.css']
```
Comment (mprahl) [inline src/app/story/storyrow/totaltimes/totaltimes.component.ts]: Could you please refactor this to not rely on `storysidebar.component.css`? Since the two components are not related, it is a bit confusing. If you'd like to share styles between them, You could either move those classes to the global `styles.css`, or use a separate shared CSS file.

File: src/app/story/storyrow/totaltimes/totaltimes.component.html [hunk 11]
```diff
@@ -0,0 +1,18 @@
...
+            <td class="timeline-data">Total processing time: </td>
+            <td class="timeline-data">{{ totalProcessingTime | timeDisplay }}</td>
+          </tr>
+          <tr class="sidebar-properties__row">
+            <td class="timeline-data">Total wait time: </td>
+            <td class="timeline-data">{{ totalWaitTime | timeDisplay }}</td>
+          </tr>
+        </tbody>
+    </table>
+</div>
```
Comment (mprahl) [inline src/app/story/storyrow/totaltimes/totaltimes.component.html]: Could you add a new line at the end of the file please?

File: src/app/story/storyrow/totaltimes/totaltimes.component.spec.ts [hunk 12]
```diff
@@ -0,0 +1,48 @@
...
+import { NoopAnimationsModule } from '@angular/platform-browser/animations';
+import { NO_ERRORS_SCHEMA } from '@angular/core';
+import { By } from '@angular/platform-browser';
+import { BsDropdownModule } from 'ngx-bootstrap/dropdown';
+
+import { TotaltimesComponent } from './totaltimes.component';
+import { TimeDisplayPipe } from '../../../pipes/timedisplay';
+
+
+describe('RecentsComponent', () => {
```
Comment (mprahl) [inline src/app/story/storyrow/totaltimes/totaltimes.component.spec.ts]: `RecentsComponent` is not correct here. Could you please updated it?

File: src/app/story/storyrow/totaltimes/totaltimes.component.spec.ts [hunk 13]
```diff
@@ -0,0 +1,48 @@
...
+        TimeDisplayPipe,
+      ],
+      providers: [],
+      imports: [
+        HttpClientTestingModule,
+        NoopAnimationsModule,
+        BsDropdownModule.forRoot()
+      ],
+      // This is used to avoid having to import components we don't test here
+      schemas: [NO_ERRORS_SCHEMA]
```
Comment (mprahl) [inline src/app/story/storyrow/totaltimes/totaltimes.component.spec.ts]: It's a pretty small component with few dependencies, is `NO_ERRORS_SCHEMA` necessary?

File: src/app/story/storyrow/totaltimes/totaltimes.component.ts [hunk 14]
```diff
@@ -0,0 +1,19 @@
...
+})
+export class TotaltimesComponent {
+
+  @Input() totalLeadTime: number;
+  @Input() totalProcessingTime: number;
+  @Input() totalWaitTime: number;
+
+  sidebarOpen = false;
+
+  constructor() {}
```
Comment (mprahl) [inline src/app/story/storyrow/totaltimes/totaltimes.component.ts]: Can this be removed?

File: src/app/story/storysidebar/storysidebar.component.css [hunk 15]
```diff
@@ -104,3 +104,14 @@
     /* Push "Properties" to the right 1rem */
     margin-right: 1rem;
 }
+
+@media (min-width: 992px) {
```
Comment (mprahl) [inline src/app/story/storysidebar/storysidebar.component.css]: This CSS seems to be unrelated. Could you please remove it if it's not needed?

File: src/app/story/storyrow/totaltimes/totaltimes.component.css [hunk 16]
```diff
@@ -0,0 +1,13 @@
+@media (max-width: 991px) {
```
Comment (mprahl) [inline src/app/story/storyrow/totaltimes/totaltimes.component.css]: Optional: In this project, we try to follow the bootstrap model of applying styles to mobile by default and then making changes to larger screens, not the other way around. For example, to follow the rest of the code base, you would do:

```css
.mobile-hide {
    display: none;
}

@media (min-width: 992px) {
    .mobile-hide {
        display: initial;
    }
}
```

File: src/app/story/story.component.ts (hunk 7 above)
Comment (sarah256) [reply src/app/story/story.component.ts]: I'm having git issues, sorry! For some reason VS code isnt switching when i switch branches on git so this accidentally got pushed

File: src/app/pipes/timedisplay.ts [hunk 17]
```diff
@@ -0,0 +1,28 @@
...
```
[the remark this answers is not in the document]
Comment (sarah256) [reply src/app/pipes/timedisplay.ts]: I'm going to do this in PR#98 if that is ok, because we refactor this entire file there anyway

File: src/app/story/story.component.html (hunk 6 above)
Comment (sarah256) [reply src/app/story/story.component.html]: I think that would look nicer, but do you think it would affect readability much? It might be less confusing if we just get the data we want straight from the source at the beginning.

File: src/app/story/storyrow/totaltimes/totaltimes.component.ts (hunk 10 above)
Comment (sarah256) [reply src/app/story/storyrow/totaltimes/totaltimes.component.ts]: It ended up that the css in `storysidebar.component.ts` wan't all needed, so I just altered it to what worked best and put it in the totaltimes css file. It let me take out some css in the storyrow css file as well.

File: src/app/story/story.component.html (hunk 6 above)
Comment (mprahl) [reply src/app/story/story.component.html]: I don't think it changes the readability. It's just less properties to manage, and if we need to use additional metadata to the story row component, we don't have to add another property.

File: src/app/story/storyrow/totaltimes/totaltimes.component.css [hunk 18]
```diff
@@ -0,0 +1,27 @@
+@media (min-width: 992px) {
```
Comment (mprahl) [inline src/app/story/storyrow/totaltimes/totaltimes.component.css]: This media query needs to be after the `display: none` or else the `display: none` always applies.

File: src/app/story/storyrow/totaltimes/totaltimes.component.css [hunk 19]
```diff
@@ -0,0 +1,27 @@
...
+    .mobile-hide {
+        display: initial;
+    }
+}
+
+.mobile-hide {
+    display: none !important;
```
Comment (mprahl) [inline src/app/story/storyrow/totaltimes/totaltimes.component.css]: Optional: Ideally, this CSS should be in the story row component applied on the entire component since this only needs to be hidden on mobile in the context of it being a column in the story row.

File: src/app/story/storyrow/totaltimes/totaltimes.component.css [hunk 20]
```diff
@@ -0,0 +1,27 @@
+@media (min-width: 992px) {
+    .mobile-hide {
+        display: initial;
+    }
+}
+
+.mobile-hide {
```
Comment (mprahl) [inline src/app/story/storyrow/totaltimes/totaltimes.component.css]: Optional: Instead of `mobile-hide` as a class name, I'd prefer a name describing what is being hidden such as `timeline-wrapper`.

File: src/app/story/storyrow/totaltimes/totaltimes.component.html [hunk 21]
```diff
@@ -0,0 +1,18 @@
+<div class="mobile-hide">
+    <table class="timeline-properties">
+        <tbody>
+          <tr class="timeline-properties__row">
+            <td class="timeline-data">Total lead time: </td>
```
Comment (mprahl) [inline src/app/story/storyrow/totaltimes/totaltimes.component.html]: Could you please remove the space after the colon? I don't think it does anything. Could you also do this for the others please?

File: src/app/story/storyrow/totaltimes/totaltimes.component.spec.ts [hunk 22]
```diff
@@ -0,0 +1,46 @@
+import { fakeAsync, ComponentFixture, TestBed } from '@angular/core/testing';
+import { HttpClientTestingModule } from '@angular/common/http/testing';
+import { NoopAnimationsModule } from '@angular/platform-browser/animations';
+import { NO_ERRORS_SCHEMA } from '@angular/core';
```
Comment (mprahl) [inline src/app/story/storyrow/totaltimes/totaltimes.component.spec.ts]: Could you please clean up this unused import?

File: src/app/story/storyrow/totaltimes/totaltimes.component.spec.ts [hunk 23]
```diff
@@ -0,0 +1,46 @@
...
+    TestBed.configureTestingModule({
+      declarations: [
+        TotalTimesComponent,
+        TimeDisplayPipe,
+      ],
+      providers: [],
+      imports: [
+        HttpClientTestingModule,
+        NoopAnimationsModule,
+        BsDropdownModule.forRoot()
```
Comment (mprahl) [inline src/app/story/storyrow/totaltimes/totaltimes.component.spec.ts]: This can be removed.

File: src/app/story/storyrow/totaltimes/totaltimes.component.spec.ts [hunk 24]
```diff
@@ -0,0 +1,46 @@
...
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      declarations: [
+        TotalTimesComponent,
+        TimeDisplayPipe,
+      ],
+      providers: [],
+      imports: [
+        HttpClientTestingModule,
+        NoopAnimationsModule,
```
Comment (mprahl) [inline src/app/story/storyrow/totaltimes/totaltimes.component.spec.ts]: This can be removed.

File: src/app/story/storyrow/totaltimes/totaltimes.component.spec.ts [hunk 25]
```diff
@@ -0,0 +1,46 @@
...
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      declarations: [
+        TotalTimesComponent,
+        TimeDisplayPipe,
+      ],
+      providers: [],
+      imports: [
+        HttpClientTestingModule,
```
Comment (mprahl) [inline src/app/story/storyrow/totaltimes/totaltimes.component.spec.ts]: This can be removed.

File: src/app/story/storyrow/totaltimes/totaltimes.component.ts [hunk 26]
```diff
@@ -0,0 +1,16 @@
...
+  templateUrl: './totaltimes.component.html',
+  styleUrls: ['./totaltimes.component.css']
+})
+export class TotalTimesComponent {
+
+  @Input() totalLeadTime: number;
+  @Input() totalProcessingTime: number;
+  @Input() totalWaitTime: number;
+
+  sidebarOpen = false;
```
Comment (mprahl) [inline src/app/story/storyrow/totaltimes/totaltimes.component.ts]: This can be removed.

File: src/app/story/storysidebar/story.sidebar.component.spec.ts [hunk 27]
```diff
@@ -57,7 +59,7 @@ describe('StorysidebarComponent testing', () => {
     expect(titleAnchorEl.href).toBe('https://bugzilla.redhat.com/show_bug.cgi?id=23456');
     const titleEl = fixture.debugElement.query(By.css('.sidebar__external-system-link > h2')).nativeElement;
     expect(titleEl.textContent).toBe('RHBZ#23456');
-    const sidebarPropertiesEl = fixture.debugElement.query(By.css('.sidebar-properties > tbody')).nativeElement;
+    const sidebarPropertiesEl = fixture.debugElement.query(By.css('.sidebar > .sidebar-properties > tbody')).nativeElement;
```
Comment (mprahl) [inline src/app/story/storysidebar/story.sidebar.component.spec.ts]: Why is this change necessary?

File: src/app/story/storysidebar/story.sidebar.component.spec.ts [hunk 28]
```diff
@@ -23,7 +24,8 @@ describe('StorysidebarComponent testing', () => {
             PropertyDisplayPipe,
             PropertyValueDisplayPipe,
             TruncatePipe,
-            NodeDisplayNamePipe
+            NodeDisplayNamePipe,
+            TimeDisplayPipe
```
Comment (mprahl) [inline src/app/story/storysidebar/story.sidebar.component.spec.ts]: Why is this change necessary?

File: src/app/story/storysidebar/story.sidebar.component.spec.ts [hunk 29]
```diff
@@ -105,7 +107,7 @@ describe('StorysidebarComponent testing', () => {
   }));
 
   it('should truncate long property values', () => {
-    const sidebarPropertiesEl = fixture.debugElement.query(By.css('.sidebar-properties > tbody')).nativeElement;
+    const sidebarPropertiesEl = fixture.debugElement.query(By.css('.sidebar > .sidebar-properties > tbody')).nativeElement;
```
Comment (mprahl) [inline src/app/story/storysidebar/story.sidebar.component.spec.ts]: Why is this change necessary?

File: src/app/story/storysidebar/story.sidebar.component.spec.ts [hunk 30]
```diff
@@ -128,7 +130,7 @@ describe('StorysidebarComponent testing', () => {
     expect(titleAnchorEl).toBeNull();
     const titleEl = fixture.debugElement.query(By.css('.sidebar > a > h3'));
     expect(titleEl).toBeNull();
-    const sidebarPropertiesEl = fixture.debugElement.query(By.css('.sidebar-properties'));
+    const sidebarPropertiesEl = fixture.debugElement.query(By.css('.sidebar > .sidebar-properties'));
```
Comment (mprahl) [inline src/app/story/storysidebar/story.sidebar.component.spec.ts]: Why is this change necessary?

File: src/app/story/storysidebar/storysidebar.component.html [hunk 31]
```diff
@@ -10,11 +10,11 @@
   <ng-container *ngIf="sidebarOpen && node">
     <a class="sidebar__external-system-link" [href]="node | nodeExternalUrl" target="_blank">
       <!-- *ngIf is used because it should only tooltip on long UIDs -->
-      <h2 *ngIf="(node.display_name | nodeDisplayName).length > 30; else shortUid" tooltip="{{ node.display_name | nodeDisplayName }}">
-        {{ node.display_name | nodeDisplayName | truncate: 30 }}
+      <h2 *ngIf="node.display_name.split(' ').slice(-1)[0].length > 30; else shortUid" tooltip="{{ node.display_name.split(' ').slice(-1)[0] }}">
```
Comment (mprahl) [inline src/app/story/storysidebar/storysidebar.component.html]: Why are these changes to the sidebar necessary?

File: src/app/story/storysidebar/storysidebar.component.ts [hunk 32]
```diff
@@ -1,4 +1,4 @@
-import { Component, OnInit, Input, OnChanges } from '@angular/core';
+import { Component, OnInit, Input, OnChanges, Output, EventEmitter } from '@angular/core';
```
Comment (mprahl) [inline src/app/story/storysidebar/storysidebar.component.ts]: Why are these changes necessary?

File: src/app/story/storysidebar/story.sidebar.component.spec.ts (hunk 27 above)
Comment (sarah256) [reply src/app/story/storysidebar/story.sidebar.component.spec.ts]: I think this was necessary before when I was doing 'inner-sidebar' but now that that's gone it should be removed.

File: src/app/story/storyrow/totaltimes/totaltimes.component.spec.ts [hunk 33]
```diff
@@ -0,0 +1,39 @@
+import { fakeAsync, ComponentFixture, TestBed } from '@angular/core/testing';
+import { HttpClientTestingModule } from '@angular/common/http/testing';
```
Comment (mprahl) [inline src/app/story/storyrow/totaltimes/totaltimes.component.spec.ts]: Please remove this unnecessary import

File: src/app/story/storyrow/totaltimes/totaltimes.component.spec.ts [hunk 34]
```diff
@@ -0,0 +1,39 @@
...
+  let fixture: ComponentFixture<TotalTimesComponent>;
+  let component: TotalTimesComponent;
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      declarations: [
+        TotalTimesComponent,
+        TimeDisplayPipe,
+      ],
+      providers: [],
```
Comment (mprahl) [inline src/app/story/storyrow/totaltimes/totaltimes.component.spec.ts]: Optional: I think you could just remove this

File: src/app/story/storyrow/totaltimes/totaltimes.component.spec.ts [hunk 35]
```diff
@@ -0,0 +1,39 @@
...
+      imports: [],
```
Comment (mprahl) [inline src/app/story/storyrow/totaltimes/totaltimes.component.spec.ts]: Optional: I think you could just remove this

File: src/app/story/storyrow/totaltimes/totaltimes.component.css [hunk 36]
```diff
@@ -0,0 +1,27 @@
+.timeline-wrapper {
+    display: none !important;
```
Comment (mprahl) [inline src/app/story/storyrow/totaltimes/totaltimes.component.css]: Remove the `!important` or else it keeps the media query below from taking effect.

**Comment by mprahl** [comment]
This is not displaying correctly on Firefox:
image

File: src/app/story/storyrow/storyrow.component.css [hunk 37]
```diff
@@ -162,3 +162,12 @@
     /* Hides the shadow effect of node's img while it is being bordered */
     opacity: 0;
 }
+
+.total-time-column {
+    position: relative;
+}
+
+.total-times {
```
Comment (mprahl) [inline src/app/story/storyrow/storyrow.component.css]: Could you please put these classes in the "Elements" section of this file? This class should be called something like `.total-time-column__total-times` to match the BEM convention.

Comment (mprahl) [inline src/app/story/storyrow/storyrow.component.css]: Could you please put these classes in the "Elements" section of this file? This class should be called something like `.total-time-column__total-times` to match the BEM convention.

File: src/app/story/storyrow/totaltimes/totaltimes.component.css [hunk 38]
```diff
@@ -0,0 +1,27 @@
...
+    display: none;
+}
+
+@media (min-width: 992px) {
+    .timeline-wrapper {
+        display: initial;
+    }
+}
+
+.timeline-data {
```
Comment (mprahl) [inline src/app/story/storyrow/totaltimes/totaltimes.component.css]: Could you rename this to match the BEM convention? Something like `timeline-properties__row__data`. This should also be at the end of the CSS file.

File: src/app/story/storyrow/storyrow.component.html [hunk 39]
```diff
@@ -65,3 +65,11 @@
     </div>
   </div>
 </div>
+<div class="node-siblings-column story-row-column total-time-column" *ngIf="first">
```
Comment (mprahl) [inline src/app/story/storyrow/storyrow.component.html]: This code isn't tested as far as I can tell. You need to add a unit test to make sure the timeline shows up on the first row.

File: src/app/story/storyrow/storyrow.component.css [hunk 40]
```diff
@@ -162,3 +167,7 @@
     /* Hides the shadow effect of node's img while it is being bordered */
     opacity: 0;
 }
+
+.total-time-column {
```
Comment (mprahl) [inline src/app/story/storyrow/storyrow.component.css]: This should be plural to match the rest `total-times-column`. This should also be in the "Blocks" section.
